shell/android-studio/reicast/src/main/java/com/reicast/emulator/emu/GL2JNIView.java

```diff
@@ -659,7 +659,7 @@ void Die() {
         }
 
         void reiosInfo(String reiosId, String reiosSoftware) {
-            if (fileName != null) {
+            if (fileName != null && !reiosId.equals("") && !reiosSoftware.equals("")) {
```

Comment:
Something I remember from my time writing Java code is that it's a good idea to reverse the equals check for an empty string to remove possible nullpointer exceptions.

``"".equals(reiosId)``

Reply:
`fileName` is the only value that initializes with a `null`

Reply:
In all reality, though, I was trying to work out the logic for how to avoid even getting here with a blank. No reason to get all the way to the shell just to throw it out.